src/main/java/net/mcreator/ui/modgui/ModElementGUI.java

```diff
@@ -500,6 +500,11 @@ private void showErrorsMessage(AggregatedValidationResult validationResult) {
 	 * This method implements the mod element saving and generation
 	 */
 	private void finishModCreation(boolean closeTab) {
+		// after mod element is updated, references to its files generated earlier may become invalid
+		// delete outdated mod element files before regeneration
+		if (editingMode)
+			mcreator.getGenerator().removeElementFiles(modElement);
```

Comment:
Is this better, @KlemenDEV?

Reply:
But this is still in UI, if one runs generator freely without UI (eg. regeneration of code) this will not run. Not ok

Reply:
May I add a cache to mod elements to store previous versions of their generatable elements?

Reply:
I can see this working, yes. You could add a new system in ModElementManager to store the previous version if there is one

Reply:
✔️ Fixed
The new system only applies to source files. As soon as this gets merged, the same hook for localization keys can be added in GUI translations tokens PR where dynamic keys support will be needed (unless we leave it to users to manage outdated translation keys).